wire-library/wire-compiler/src/main/java/com/squareup/wire/WireCompiler.kt

```diff
@@ -143,66 +136,43 @@ class WireCompiler internal constructor(
       )
     }
 
-    Closer.create().use { closer ->
-      val sources = protoPaths.map { it.toPath() }
-      val directories = directoryPaths(closer, sources)
+    val directories = protoPaths.map { it.toPath() }
 
-      val allDirectories = directories.map { Location.get(it.key.toString()) }.toList()
-      val sourcePath: List<Location>
-      val protoPath: List<Location>
-
-      if (sourceFileNames.isNotEmpty()) {
-        sourcePath = sourceFileNames.map { locationOfProto(directories, it) }
-        protoPath = allDirectories
-      } else {
-        sourcePath = allDirectories
-        protoPath = listOf()
-      }
+    val allDirectories = directories.map { Location.get(it.toString()) }.toList()
+    val sourcePath: List<Location>
+    val protoPath: List<Location>
 
-      val wireRun = WireRun(
-          sourcePath = sourcePath,
-          protoPath = protoPath,
-          treeShakingRoots = treeShakingRoots,
-          treeShakingRubbish = treeShakingRubbish,
-          targets = targets,
-          modules = modules,
-          permitPackageCycles = permitPackageCycles
-      )
-
-      wireRun.execute(fs, log)
+    if (sourceFileNames.isNotEmpty()) {
+      sourcePath = sourceFileNames.map { locationOfProto(directories, it) }
+      protoPath = allDirectories
+    } else {
+      sourcePath = allDirectories
+      protoPath = listOf()
     }
-  }
 
-  /**
-   * Map the physical path to the file system root. For regular directories the key and the
-   * value are equal. For ZIP files the key is the path to the .zip, and the value is the root
-   * of the file system within it.
-   */
-  private fun directoryPaths(closer: Closer, sources: List<Path>): Map<Path, Path> {
-    val directories = mutableMapOf<Path, Path>()
-    for (source in sources) {
-      directories[source] = when {
-        Files.isRegularFile(source.toNioPath()) -> {
-          val sourceFs = FileSystems.newFileSystem(source.toNioPath(), javaClass.classLoader)
-          closer.register(sourceFs)
-          sourceFs.rootDirectories.single().toOkioPath()
-        }
```

Comment:
Not sure why we were doing all this magic

Reply:
This opens Zips!